src/cladetime/assign_clades.py

```diff
@@ -151,25 +124,40 @@ def merge_metadata(config: Config) -> pl.DataFrame:
     default=None,
     help="Directory where the clade assignment file will be saved. Default: [home dir]/covid_variant/",
 )
-def main(sequence_released_since_date: datetime.date, reference_tree_date: datetime.date, data_dir: str | None):
-    # TODO: do we need additional date validations (e.g., no future dates)?
+@time_function
+def main(sequence_collection_date: datetime.date, tree_as_of: datetime.date, data_dir: str | None):
+    # TODO: update CLI options as discussed, including an option to save the linefile
 
-    config = setup_config(data_dir, sequence_released_since_date, reference_tree_date)
-    logger.info("Starting pipeline", reference_tree_date=reference_tree_date, run_time=config.run_time)
+    config = _setup_config(data_dir)
+    logger.info("Starting pipeline", reference_tree_date=tree_as_of, run_time=config.run_time)
 
     os.makedirs(config.data_path, exist_ok=True)
-    nextclade_dataset_path = get_nextclade_dataset(config.reference_tree_date, config.data_path)
-    get_sequences(config)
-    get_sequence_metadata(config)
-    assign_clades(config, nextclade_dataset_path)
-    merged_data = merge_metadata(config)
-    merged_data.write_csv(config.assignment_file)
+
+    with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
Unlike prior versions of `assign_clade`, this new thinking assumes that we can save the intermediate files required for clade assignment in a temporary directory and only worry about saving the final output(s) to disk.